lightning/src/ln/features.rs

```diff
@@ -19,28 +20,96 @@ mod sealed { // You should just use the type aliases instead.
 	impl Context for NodeContext {}
 	impl Context for ChannelContext {}
 
-	pub trait DataLossProtect: Context {}
-	impl DataLossProtect for InitContext {}
-	impl DataLossProtect for NodeContext {}
-
-	pub trait InitialRoutingSync: Context {}
-	impl InitialRoutingSync for InitContext {}
-
-	pub trait UpfrontShutdownScript: Context {}
-	impl UpfrontShutdownScript for InitContext {}
-	impl UpfrontShutdownScript for NodeContext {}
-
-	pub trait VariableLengthOnion: Context {}
-	impl VariableLengthOnion for InitContext {}
-	impl VariableLengthOnion for NodeContext {}
+	/// Defines a feature with the given bits for the specified [`Context`]s. The generated trait is
+	/// useful for manipulating feature flags.
+	///
+	/// [`Context`]: trait.Context.html
+	macro_rules! define_feature {
+		($even_bit: expr, $odd_bit: expr, $feature: ident, [$($context: ty),+], $doc: expr) => {
```

Comment:
We can probably skip specifying `$odd_bit` - its always `$even_bit + 1` (and you rely on it kinda being so by using a common `BYTE_OFFSET`

Reply:
Hmm, I don't think `initial_routing_sync` "doesn't have" an even bit, more like it doesn't really make sense to ever set it. We should handle it fine if someone sets it - we understand what's going on and will send an initial_routing_sync.

Reply:
From our offline discussion, we decided to make the behavior change.